### ISSUE TITLE
Wagtail react streamfield optimization

### ISSUE THREAD

**Issue by joeyjurjens** [coderedcorp/coderedcms#369]
I just stumbled upon the [wagtail react streamfield editor](https://github.com/wagtail/wagtail-react-streamfield) and tried installing it on a coderedcms project.

It basically already works, though I wonder if coderedcms may have some optimization for this? I'm ok with doing this myself, but I'm not sure if you guys are open for it.

The optimizations I'd like to make would be:

- Auto collapse all blocks based on user preference (or default?); Streamfields in wagtail get big and unmanageable very quickly in my opinion and the collapse option reduces this issue.
- Fix the advanced settings button, as it's broken now

and possibly some other things if I find any.

No is an answer, then a separate package would be an option for me to make.
But I think this streamfield editor can come in very handy with coderedcms!

Hope to hear from you.

**Comment by vsalvino** [comment]
Thanks for the suggestion Joey. Having the react streamfield collapse all block by default would probably be a huge improvement, as coderedcms is very block-heavy and the main complaint is that it is difficult to edit/navigate the streamfield on a big page. We were hoping to switch to the react streamfield once it is merged into wagtail, but if you are able to get it working now, that would be a huge help!

We would certainly welcome a pull request if any improvements are required in coderedcms to make this happen. Let me know if there's any way I can help or answer questions.

**Comment by vsalvino** [comment]
Flagging needs research, based on comments related to Wagtail 2.13 in the pull request.

**Comment by joeyjurjens** [comment]
Closing because of; #372
